meson.build

```diff
@@ -0,0 +1,72 @@
+project('ladybird-gtk', 'c', 'cpp',
+    license: 'BSD-2-Clause',
+    version: '1.0-dev',
+    default_options: [ 'warning_level=1', 'buildtype=debugoptimized', 'cpp_std=c++20' ],
+)
+
+gnome = import('gnome')
+
+gtk_min_version = '>= 4.11.3'
+libadwaita_min_version = '>=1.4'
+
+gtk_dep = dependency('gtk4', version: gtk_min_version, default_options: [ 'build-tests=false', 'build-testsuite=false' ])
+libadwaita_dep = dependency('libadwaita-1', version: libadwaita_min_version)
+
+lagom_dep = dependency('Lagom', method: 'cmake', 
+  modules: [
+    'Lagom::Core',
+    'Lagom::FileSystem',
+    'Lagom::Gfx',
+    'Lagom::IPC',
+    'Lagom::Main',
+    'Lagom::Web',
+    'Lagom::WebView',
+    'Lagom::SQL',
+    'Lagom::Protocol'
+  ]
+)
+ladybird_dep = dependency('ladybird', method: 'cmake',
+   modules: [ 'ladybird::ladybird' ]
+)
+
+add_project_arguments([
+    '-DAK_DONT_REPLACE_STD',
+    '-Wno-literal-suffix',
+], language: 'cpp')
+
+# FIXME: Can't figure out how to get this guy to pull the typelibs from the subprojects
```

Comment:
This now works assuming you BYOL, except that I can't figure out how to get the blueprint-compiler to look in the subdirs for the updated libadwaita-1 typelibs.

```
cmake -GNinja -S ../serenity/Meta/Lagom -B build-lagom -DCMAKE_PREFIX_PATH=install-lagom -DBUILD_LAGOM=ON -DENABLE_LAGOM_LADYBIRD=ON -DENABLE_QT=OFF
cmake --build build-lagom
cmake --install build-lagom
meson setup --cmake-prefix-path=$PWD/install-lagom build
ninja -C build
```
```
[16/20] Generating blueprints with a custom command (wrapped by meson to set env)
FAILED: window.ui 
/home/andrew/.local/bin/meson --internal exe --unpickle /home/andrew/ladybird-gtk4/build/meson-private/meson_exe_blueprint-compiler_65359fc1c5b59290615bf54a9fed7aa41d291792.dat
while executing ['/home/andrew/ladybird-gtk4/build/subprojects/blueprint-compiler/blueprint-compiler', 'batch-compile', '.', '../', '../window.blp']
--- stdout ---
error: Namespace Adw does not contain a type called TabOverview
at ../window.blp line 8 column 7:
   8 |  Adw.TabOverview tab_overview {
     |      ^
hint: Did you check your spelling?
hint: Are your dependencies up to date?
```

Reply:
nope, it doesn't like getting the libdir

 `../meson.build:37:0: ERROR: Could not get an internal variable and no default provided for <InternalDependency null: True>`

Reply:
well, I got it working, but it's not pretty.

Reply:
Yeah that's not pretty. Don't forget that libadwaita could be coming from an existing installation (not being built as a subproject), or, if Ladybird itself is being built as someone's subproject, the `subprojects` directory would be at the top level and not here.

Hm. We could at least try to detect whether libadwaita is an external or internal (subproject) dependency, and do different things depending on that.